pinot-core/src/main/java/org/apache/pinot/core/operator/filter/FilterOperatorUtils.java

```diff
@@ -54,6 +54,12 @@ public static BaseFilterOperator getLeafFilterOperator(PredicateEvaluator predic
 
     Predicate.Type predicateType = predicateEvaluator.getPredicateType();
 
+    //Only for dictionary encoded columns and offline data sources
+    if (predicateType == Predicate.Type.RANGE && dataSource.getDictionary() != null
```

Comment:
Why only dictionary encoded? For raw columns, each range could be a range of start and end raw values instead of dictIds. The data structure could be sorted on start raw values (just like it is on start dictIds)

Reply:
Let's add a TODO